### ISSUE TITLE
Feature request: Don't make Api an inner interface of ApiClient for feign

### ISSUE THREAD

**Issue by cod3hulk** [swagger-api/swagger-codegen#5908]
##### Description

If you only want to have minimal dependencies for your project by disabling `generateSupportingFiles` this is currently not really possible due to the internal `Api` interface included in the `ApiClient` which is used for generating the Feign Client interfaces as extension. 

##### Swagger-codegen version

2.2.3-SNAPSHOT

##### Suggest a Fix

Either remove the entire `Api` interface from the Feign Client interface, because it is empty. If this interface is somehow needed as marker interface it would be great if it could be an extra class of its own - so it could be added to the `supportingFilesToGenerate` to keep the number of needed dependencies as small as possible.

**Comment by wing328** [comment]
@cod3hulk can you please elaborate a bit more on the minimal dependencies? What do you want to achieve? Would generating `feign` client with the `-t` option using customized templates meet your requirement?

**Comment by cbornet** [comment]
That interface looks useless to me. It should be removed.

**Comment by wing328** [comment]
cc @bbdouglas who has been doing a lot of works with the Java Feign generator.

**Comment by bbdouglas** [comment]
I agree that class looks superfluous, especially since it's empty. I'm guessing it was added for future-compatibility's sake, in case someone wanted to add methods in the future that all Feign api's support, such as setting timeouts. But that's just a guess. I don't see much use for it, and I don't see why anyone would directly reference that inner interface since it's empty.

It looks safe to delete from my perspective. Since it's a public interface, it does represent a possible break in backwards compatibility to remove it, though. If that's a concern, we could deprecate it now and remove in the next major release.

**Comment by cod3hulk** [comment]
@bbdouglas sounds good, would be great if we could get rid of this interface.

@wing328 We have to add the following dependencies in our project, which are needed by the ApiClient:
* org.apache.oltu.oauth2:org.apache.oltu.oauth2.client
* com.fasterxml.jackson.datatype:jackson-datatype-joda
* io.github.openfeign.form:feign-form
* io.github.openfeign:feign-slf4j
* io.github.openfeign:feign-jackson

Our basic usage/use case is providing swagger.json files to our consuming services to be able to generate Feign client interfaces. So the ApiClient is not needed for our scenario. We want to keep the configuration/dependencies as little as possible to make the integration/generation of swagger FeignClients in our services as simple as possible.
Haven't tested the `-t` option so far, but will give it a try. Nevertheless, I think the generated feign client should only depend on a minimum set of needed classes.
